File: cpp-frame/udb/tidAndProb.cpp
#include "tidAndProb.h"

using namespace std;

tidAndProb::tidAndProb(int tid, double probability){
	this->tid = tid;
	this->probability = probability;
}

tidAndProb::~tidAndProb(){

}

int tidAndProb::getTid(){
	return tid;
}

double tidAndProb::getProbability(){
	return probability;
}

File: code/input/transaction_database_list_integers/TidAndProb.java
package code.input.transaction_database_list_integers;

public class TidAndProb{
    private int tid;
    private double probability;

    public TidAndProb (int tid, double probability){
        this.tid = tid;
        this.probability = probability;
    }

    public int getTid(){
        return this.tid;
    }

    public double getProbability(){
        return this.probability;
    }

    public String toString(){
        return "" + tid + " (" + probability + ")";
    }

    public int hashCode() {
        String string = "" + tid;
        return string.hashCode();
    }
}